### ISSUE TITLE
Bug with `%Nd` substitution for negative numbers in GDScript

### ISSUE THREAD

**Issue by dalexeev** [godotengine/godot#42413]
**Godot version:**
3.2, 4.0

**OS/device including version:**
All

**Issue description:**
The string formatting operator does not work correctly for `%Nd` substitution if the number is negative (it does not take into account that the minus sign occupies one character).

**Steps to reproduce:**
```gdscript
tool
extends EditorScript

func _run():
    print("%3.0f" % 5)
    print("%3.0f" % -5)
    print("%3d" % 5)
    print("%3d" % -5)
```

```
  5
 -5
  5
  -5
```

**Minimal reproduction project:**
—

**Comment by ssw99** [comment]
I want to fix this bug, it seems easy

**Comment by dalexeev** [comment]
The bug is reproducible in 4.0 (`4.0.dev.calinou.4a4be0626`). Please add milestone `4.0`.

Do we need to fix the bug in 3.2 (formally, this is a violation of compatibility with 3.2.3)?

@ssw99 Feel free to open PR.
